src/SWP/Bundle/CoreBundle/Controller/FbApplicationController.php

```diff
@@ -0,0 +1,131 @@
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\ConflictHttpException;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+class FbApplicationController extends Controller
+{
+    /**
+     * @ApiDoc(
+     *     resource=true,
+     *     description="Lists Facebook Applications",
+     *     statusCodes={
+     *         200="Returned on success.",
+     *         500="Unexpected error."
+     *     }
+     * )
+     * @Route("/api/{version}/facebook/applications/", options={"expose"=true}, defaults={"version"="v1"}, name="swp_api_list_facebook_applications")
+     * @Method("GET")
+     */
+    public function listAction(Request $request)
+    {
+        $repository = $this->get('swp.repository.facebook_application');
+
+        $items = $repository->getPaginatedByCriteria(
+            new Criteria(),
+            ['id' => 'asc'],
+            new PaginationData($request)
+        );
+
+        return new ResourcesListResponse($items);
+    }
+
+    /**
+     * @ApiDoc(
+     *     resource=true,
+     *     description="Create Facebook application",
+     *     statusCodes={
+     *         201="Returned on success.",
+     *         400="Returned when not valid data."
+     *     },
+     *     input="SWP\Bundle\CoreBundle\Form\Type\FacebookApplicationType"
+     * )
+     * @Route("/api/{version}/facebook/applications/", options={"expose"=true}, defaults={"version"="v1"}, name="swp_api_create_facebook_applications")
+     * @Method("POST")
+     */
+    public function createAction(Request $request)
+    {
+        /* @var FacebookApplication $feed */
+        $application = $this->get('swp.factory.facebook_application')->create();
+        $form = $this->createForm(FacebookApplicationType::class, $application, ['method' => $request->getMethod()]);
+
+        $form->handleRequest($request);
+        if ($form->isValid()) {
+            $this->checkIfThatApplicationAlreadyExists($application);
+            $this->get('swp.repository.facebook_application')->add($application);
+
+            return new SingleResourceResponse($application, new ResponseContext(201));
+        }
+
+        return new SingleResourceResponse($form, new ResponseContext(400));
+    }
+
+    /**
+     * @ApiDoc(
+     *     resource=true,
+     *     description="Delete Facebook application",
+     *     statusCodes={
+     *         204="Returned on success.",
+     *         500="Unexpected error.",
+     *         404="Application not found",
+     *         409="Application is used by page"
+     *     }
+     * )
+     * @Route("/api/{version}/facebook/applications/{id}", options={"expose"=true}, defaults={"version"="v1"}, name="swp_api_delete_facebook_applications")
+     * @Method("DELETE")
+     */
+    public function deleteAction($id)
+    {
+        if (null === $application = $this->get('swp.repository.facebook_application')->findOneBy(['id' => $id])) {
```

Comment:
`$this->get('swp.repository.facebook_application')` can be assigned to variable as this service is used more than once in the same method